hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecordGlobalLocation.java

```diff
@@ -67,8 +67,7 @@ public boolean equals(Object o) {
     HoodieRecordGlobalLocation otherLoc = (HoodieRecordGlobalLocation) o;
     return Objects.equals(partitionPath, otherLoc.partitionPath)
         && Objects.equals(instantTime, otherLoc.instantTime)
-        && Objects.equals(fileId, otherLoc.fileId)
-        && Objects.equals(position, otherLoc.position);
+        && Objects.equals(fileId, otherLoc.fileId);
```

Comment:
are there any side effects wrt positional deletes if we remove position from equals? Also, don't we need to keep the same fields in hashcode as well?

Reply:
The hashCode function does not use `position` at the first place. Do not use any use case. `HoodieRecordLocation` has the same logic.

Reply:
If the position is not even used anywhere, we better remove it from the class, the only usage is from `HoodieKeyLocationFetchHandle.globalLocations`, where the `getHoodieKeyIterator` should be used instead of `fetchRecordKeysWithPositions`.

Reply:
I meant the position is used from this class, but there should not be any cases where we need to group records at position level.

Reply:
I didn't see any usesage of the `position` outside this class, can you confirm that?

Reply:
`HoodieGlobalSimpleIndex#tagLocationInternal` uses `fetchRecordGlobalLocations` which put the position of records into the `HoodieRecordGlobalLocation` instances.  Such instance is later put into the corresponding `HoodieRecord` instance if the record is an update.